### PR TITLE
Flat binary search over APU

### PR DESCRIPTION
**Pull request by EitanZ-GSI**
Signed-off-by: ezeharia <ezeharia@gsitechnology.com>

Explain the problem that this commit is solving. Focus on why you
are making this change as opposed to how (the code explains that).
Are there side effects or other unintuitive consequences of this
change? Here's the place to explain them.

solve issue Support search over APU board #5202
support Tanimoto/Hamming Flat binary search over APU device .
Search over APU can be done for FAISS_BIN_IDMAP engaine type only.
Dependencies : build with FPGA (- a ) and APU (-w ) . build must contain FPGA flag .
example : ./build.sh -a -w
*** search is not posible without APU device conneted to a server .**
Resolves: #5202

### CODE REVIEW

File: core/src/db/engine/ExecutionEngineImpl.cpp [hunk 1]
```diff
@@ -636,6 +640,34 @@ ExecutionEngineImpl::CopyToFpga() {
...
+Status
+ExecutionEngineImpl::CopyToApu(uint32_t row_count) {
+#ifdef MILVUS_APU_VERSION
+
+    auto cache_index_ =
+        std::static_pointer_cast<knowhere::VecIndex>(cache::FpgaCacheMgr::GetInstance()->GetItem(location_));
+    bool already_in_cache = (cache_index_ != nullptr);
+
+    if (!already_in_cache) {
+        cache::FpgaCacheMgr::GetInstance()->ClearCache();  // clear cache to support cache switch .
```
Comment (shengjun1985) [inline core/src/db/engine/ExecutionEngineImpl.cpp:652]: It seems that you want to cache only one segment index, isn't it?

Comment (EitanZ-GSI) [reply core/src/db/engine/ExecutionEngineImpl.cpp:652]: Hi , yes . 
Currntly the use of FpgaCachch is to hold the latest data set loaded to Apu according location .we are supporting load of 1 data set at a time so no need to hold more then one index. 
other funcionality is been managed by Apu libs.

**Review by yhmo — APPROVED** [review]

**Review by shengjun1985 — APPROVED** [review]

**Review by shengjun1985 — APPROVED** [review]

**Review by wxyucs — APPROVED** [review]
/approve

**Pull request merged**